design/ceph/submariner-globalnet.md

```diff
@@ -0,0 +1,96 @@
+# Submariner Globalnet support in Rook
+
+
+## Use case
+As a rook-ceph user, I should be able to setup mirroing across multiple clusters with overlapping networks in order to safegurd my data in case of any disaster.
+
+
+## Background
+[Submariner](https://submariner.io/getting-started/) provides cross-cluster connectivity between two kubernetes clusters when they have different CIDRs. Many a times customers don't have control over the network configuration of the clusters, resulting in clusters with overlapping CIDRs (ServiceCIDR and ClusterCIDR). For such scenarios, Submariner provides another tool called [Globalnet](https://submariner.io/getting-started/architecture/globalnet/). It provides cross-cluster connectivity from pods to remote services using their global IPs.
+
+In order to support kubernetes clusters connected using Submariner-Globalnet, Rook needs to use the global IPs provided by Globalnet instead of the service cluster IP.
+
+
+## Prerequisite
+- The peer clusters should be connected using Submariner.
+- Globalnet controller should be running on both the clusters.
+(Refer Submariner [documentation](https://submariner.io/getting-started/quickstart/) for installation steps on different environments)
+
+## Service of OSD pods
+- Implement k8s service for each osd pod:
```

Comment:
Explaining why this is needed will help any upstream users who aren't aware of what we have previously researched.

```suggestion
## Service of OSD pods
For scenarios like RBD mirroring, peers need direct access to Ceph OSDs. Each OSD will have to have a standard ClusterIP Service created for it to allow this. The OSD Service will be created only when multi-cluster networking support is enabled.

The reference implementation used for development of this feature is (Submariner)[link]. In the reference implementation, it is important for Services to be of `type` ClusterIP. Headless Services don't have internal routing between OSDs local to a cluster -- or any other Ceph daemons local to the cluster.
```

@travisn should we always enable OSD services or only for MCS?